tests/unit/chunker/test_markdown_chunker.py

```diff
@@ -127,7 +127,7 @@ def expected_chunks(documents):
                      '\ntext in level 3\n#### Level 4\ntext in level 4\n##### Level 5'
                      '\ntext in level 5\n###### Level 6\ntext in level 6',
                 source='doc_1',
-                metadata={'test': '1'},
```

Comment:
Why was this cast to `str` before? (Seems like we had a bug?)

Reply:
Kinda bug of Union parsing in pydantic v1. In pydantic v2 there is smart unions so they fixed this problem.